### PR TITLE
CP-25329: Refine API reference for PUSB & VUSB in xenadmin

### PR DESCRIPTION
**Pull request by makunterry**
Change points from xapi:
1. Remove "attached" field in PUSB & VUSB.
2. Add "currently_attached" as a bool field in VUSB.

Signed-off-by: Kun Ma <kun.ma@citrix.com>

### CODE REVIEW

**Comment by huizh** [comment]
@MihaelaStoica, please could you take a look at this PR?

File: XenAdmin/TabPages/UsbPage.cs [hunk 1]
```diff
@@ -239,7 +239,8 @@ private void setVm()
             {
                 if (_pusb != null)
                 {
-                    VUSB vusb = _pusb.Connection.Resolve(_pusb.attached);
+                    USB_group usbgroup = _pusb.Connection.Resolve(_pusb.USB_group);
+                    VUSB vusb = usbgroup.VUSBs.Count > 0 ? _pusb.Connection.Resolve(usbgroup.VUSBs[0]) : null;
```
Comment (MihaelaStoica) [inline XenAdmin/TabPages/UsbPage.cs]: We should check that `usbgroup` is not null before using it

**Comment by kc284** [comment]
One small remark regarding style: in C# it's preferable to use pascal case instead of camel case for methods, i.e. SetVm instead of setVm. Have a look here https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/capitalization-conventions. This document talks about public methods, but private methods follow the same conventions, with the exception perhaps of the designer autogenerated event handlers. (The API bindings do not follow the case conventions either, but it is impractical to re-write them).

**Comment by makunterry** [comment]
@MihaelaStoica I have refined the UsbPage.cs. Please take a look then.

File: XenAdmin/Dialogs/AttachUsbDialog.cs [hunk 2]
```diff
@@ -81,24 +81,32 @@ private void BuildList()
...
                         if (pusb.passthrough_enabled && !attached)
                         {
                             UsbItem usbNode = new UsbItem(pusb);
-                            treeUsbList.AddChildNode(hostNode, usbNode);
+                            usbNodeList.Add(usbNode);
                         }
                     }
                     // Show host node only when it contains available USB devices.
-                    if (hostNode.ChildNodes.Count > 0)
+                    if (usbNodeList.Count > 0)
```
[the remark this answers is not in the document]
Comment (makunterry) [reply XenAdmin/Dialogs/AttachUsbDialog.cs]: You are right.
I have refined and updated this PR.

**Pull request merged**